src/OpenLoco/src/Vehicles/Routing.cpp

```diff
@@ -683,19 +700,220 @@ namespace OpenLoco::Vehicles
+            if (trackSize.rotationEnd < 12)
+            {
+                reverseInterest.loc -= World::Pos3{ World::kRotationOffset[trackSize.rotationEnd], 0 };
+            }
+            reverseInterest.trackAndDirection ^= (1 << 2); // Reverse flag
+
+            hashMap->tryAdd(reverseInterest);
+        }
+
+        auto* trackObj = ObjectManager::get<TrackObject>(trackObjectId);
+        bool placementFailure = false;
+
+        for (auto i = 0; i < 4; ++i)
+        {
+            if (!(trackModObjectIds & (1U << i)))
+            {
+                continue;
+            }
+
+            auto* trackModObj = ObjectManager::get<TrackExtraObject>(trackObj->mods[i]);
+
+            const auto pieceFlags = TrackData::getTrackCompatibleFlags(interest.tad().id());
+            if ((trackModObj->trackPieces & pieceFlags) != pieceFlags)
+            {
+                //_1135F64 |= (1 << 0); placement failed at least once
+                placementFailure = true;
+                break;
+            }
+        }
+
+        if (!placementFailure)
+        {
+            auto trackStart = interest.loc;
+            const auto tad = interest.tad();
+            if (tad.isReversed())
+            {
+                auto& trackSize = World::TrackData::getUnkTrack(tad._data);
+                trackStart += trackSize.pos;
+                if (trackSize.rotationEnd < 12)
+                {
+                    trackStart -= World::Pos3{ World::kRotationOffset[trackSize.rotationEnd], 0 };
+                }
+            }
+
+            for (auto& trackPiece : TrackData::getTrackPiece(tad.id()))
+            {
+                auto trackLoc = trackStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ trackPiece.x, trackPiece.y }, tad.cardinalDirection()), 0 };
+                trackLoc.z += trackPiece.z;
+
+                auto tile = TileManager::get(trackLoc);
+                World::TrackElement* elTrack = nullptr;
+                for (auto& el : tile)
+                {
+                    elTrack = el.as<World::TrackElement>();
+                    if (elTrack == nullptr)
+                    {
+                        continue;
+                    }
+                    if (elTrack->baseHeight() != trackLoc.z)
+                    {
+                        continue;
+                    }
+                    if (elTrack->unkDirection() != tad.cardinalDirection())
+                    {
+                        continue;
+                    }
+                    if (elTrack->sequenceIndex() != trackPiece.index)
+                    {
+                        continue;
+                    }
+                    if (elTrack->trackObjectId() != trackObjectId)
+                    {
+                        continue;
+                    }
+                    if (elTrack->trackId() != tad.id())
+                    {
+                        continue;
+                    }
+
+                    break;
+                }
```

Comment:
this all looks identical to the anonymous lamda in CreateTrackMod.cpp - can we put it in a common place somewhere?

Reply:
In OpenRCT2 we ended up with lots of small variations of getTrackElement and i really didn't like how it was all so arbitrary as to what one you should use when. I'm not sure how we want to do it in OpenLoco. A lot of the time you want to iterate all of a certain type with some other params as well which is what #1403 was looking at. I guess we probably want a very specific getter as well that will always get a certain unique track.

Reply:
I think I'll look into making some standalone functions for this as a follow on PR.